commit.go

```diff
@@ -137,6 +137,30 @@ func (c *Commit) History() ([]*Commit, error) {
 	return commits, err
 }
 
+// Encode transforms a Commit into a core.Object.
+func (b *Commit) Encode(o core.Object) error {
+	o.SetType(core.CommitObject)
+
+	w, err := o.Writer()
+	if err != nil {
+		return err
+	}
+	w.Write([]byte("tree "))
+	w.Write([]byte(b.tree.String()))
+	for _, parent := range b.parents {
+		w.Write([]byte("\nparent "))
+		w.Write([]byte(parent.String()))
+	}
+	w.Write([]byte("\nauthor "))
+	b.Author.Encode(w)
+	w.Write([]byte("\ncommitter "))
```

Comment:
will it be better to use templates for this task?, not only here, but in the tag, blob...